### PR TITLE
Support for environments without a default VPC

### PR DESCRIPTION
**Pull request by maclema**
Allows passing a VPC and subnet ID to be used for the image build infrastructure to allow for launching in an environment where no default VPC exists.

Of note, the subnet passed in as a parameter should have internet access (i.e private subnet with NAT gateway routes, or a public subnet). When a public subnet is used, it must have MapPublicIpOnLaunch enabled as image builder won't automatically map a public IP address and the image build will timeout.

### Tests
Manually launched parallel cluster (via parallel cluster manager) stacks using a stack referencing stack set templates with these updates which exist in my personal AWS account S3 bucket.

### CODE REVIEW

File: api/infrastructure/parallelcluster-api.yaml [hunk 1]
```diff
@@ -964,6 +975,13 @@ Resources:
       Roles:
         - !Ref ImageBuilderInstanceRole
 
+  InfrastructureConfigurationSecurityGroup:
+    Condition: NonDefaultVpc
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      VpcId: !Ref ImageBuilderVpcId
+      GroupDescription: Parallel cluster image builder security group
```
Comment (enrico-usai) [inline api/infrastructure/parallelcluster-api.yaml]: Parallel cluster --> ParallelCluster :)

File: api/infrastructure/parallelcluster-api.yaml [hunk 2]
```diff
@@ -74,6 +74,14 @@ Parameters:
       - true
       - false
 
+  ImageBuilderVpcId:
+    Description: Optional. Provide a specific vpc id to use for building the container images. Use this if you don't have a default vpc available.
+    Type: String
+
+  ImageBuilderSubnetId:
+    Description: Optional. Provide a specific subnet id to use for building the container images. Use this if you don't have a default vpc available.
+    Type: String
```
Comment (demartinofra) [inline api/infrastructure/parallelcluster-api.yaml]: Could you make the `Default: ''` explicit for these 2 new params?

**Review by demartinofra — COMMENTED** [review]
Thanks for adding this support.

Could you please add a line to the changelog, something like:

```
**ENHANCEMENTS**
- Add support to deploy API infrastructure in environments without a default VPC.
```

**Comment by maclema** [comment]
Added defaults and changelog entry.

**Review by enrico-usai — APPROVED** [review]
Thanks @maclema for the improvement!

**Pull request merged**